src/VirtualClient/VirtualClient.Actions/SPECview/SpecViewExecutor.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Abstractions;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Authentication;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Platform;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+    using VirtualClient.Contracts.Metadata;
+
+    /// <summary>
+    /// Executes the 3DMark workload.
+    /// </summary>
+    [WindowsCompatible]
+
+    public class SpecViewExecutor : VirtualClientComponent
+    {
+        private IFileSystem fileSystem;
+        private ISystemManagement systemManagement;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public SpecViewExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+            : base(dependencies, parameters)
+        {
+            this.fileSystem = dependencies.GetService<IFileSystem>();
+            this.systemManagement = dependencies.GetService<ISystemManagement>();
+        }
+
+        /// <summary>
+        /// The command line argument defined in the profile.
+        /// </summary>
+        public string CommandArguments
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SpecViewExecutor.CommandArguments));
+            }
+        }
+
+        /// <summary>
+        /// The path to the RunViewperf.exe.
+        /// </summary>
+        public string ExecutablePath { get; set; }
+
+        /// <summary>
+        /// The path for the intermediate results file.
+        /// </summary>
+
+        /// <summary>
+        /// Defines the path to the SPECview package that contains the workload
+        /// executable.
+        /// </summary>
+        protected DependencyPath Package { get; set; }
+
+        /// <summary>
+        /// Executes the SPECview workload.
+        /// </summary>
+        protected override Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            return this.ExecuteWorkloadAsync(telemetryContext, cancellationToken);
```

Comment:
You can move the private method inside instead of making a private method with the same name